rcp/org.testeditor.rcp4.tcltestrun/src/org/testeditor/rcp4/tcltestrun/TclLauncher.xtend

```diff
@@ -1,131 +1,16 @@
-		return false
-	}
-
-	private def boolean launchGradleBasedTest(IProject project, String elementId) {
-		logger.info("Trying to launch gradle test execution for test {} in project {}", elementId, project)
-		progressRunner.run([ monitor |
-			monitor.beginTask("Test execution: " + elementId, IProgressMonitor.UNKNOWN)
-			val testResultFile = project.createOrGetDeepFolder(TclLauncher.GRADLE_TEST_RESULT_FOLDER).getFile(
-				elementId.elementIdToFileName).location.toFile
-			val GradleConnector connector = GradleConnector.newConnector
-			var ProjectConnection connection = null
-			try {
-				val projectFolder = project.location.makeAbsolute.toFile
-				connection = connector.forProjectDirectory(projectFolder).connect();
-				// val BuildEnvironment environment = connection.model(BuildEnvironment).get();
-				connection.newBuild.addProgressListener(new ProgressListener {
-
-					override statusChanged(ProgressEvent event) {
-						logger.info("Gradle build event: {}", event.displayName)
-					}
-
-				}) // .forTasks("test") // does not work, see issue below
-				.withArguments("test", "--tests", elementId) // https://issues.gradle.org/browse/GRADLE-2972
-				// .setStandardOutput(System.out) // alternatively get a separate console output stream (see http://wiki.eclipse.org/FAQ_How_do_I_write_to_the_console_from_a_plug-in%3F)
-				.run(
-					new ResultHandler {
-
-						override onComplete(Object obj) {
-							testResultFile.showTestResult
-						}
-
-						override onFailure(GradleConnectionException exception) {
-							logger.
-								error('''Caught error during gradle task "test" of element "«elementId»": «exception.toString»''',
-									exception)
-							project.refreshLocal(IProject.DEPTH_INFINITE, new NullProgressMonitor())
-							testResultFile.showTestResult
-						}
-
-					})
-			} finally {
-				if (connection != null) {
-					connection.close
-				}
-			}
-			monitor.done
-		])
-
-		return true
-	}
-
-	private def boolean launchMavenBasedTest(IStructuredSelection selection, IProject project, String elementId) {
-		logger.info("Trying to launch maven test execution for test {} in project {}", elementId, project)
-
-		progressRunner.run([ monitor |
-			monitor.beginTask("Test execution: " + elementId, IProgressMonitor.UNKNOWN)
-			val result = mavenExecutor.executeInNewJvm("integration-test", project.location.toOSString,
-				"test=" + elementId)
-			val testResultFile = project.createOrGetDeepFolder(TclLauncher.MVN_TEST_RESULT_FOLDER).getFile(
-				elementId.elementIdToFileName).location.toFile
-			if (result == 0) {
-				project.refreshLocal(IProject.DEPTH_INFINITE, monitor)
-				testResultFile.showTestResult
-			} else {
-				logger.error('''Error during maven task "integration-test" of element "«elementId»"''')
-			}
-			monitor.done
-		])
-		return true
-	}
-
-}
+	public def Map<String, Object> launchTest(IStructuredSelection selection, IProject project, String elementId,
```

Comment:
This method should not return a map, instead return an object of some type like `LaunchResult`.